### PR TITLE
refactor(carto): Add @carto/api-client

### PR DESCRIPTION
**Pull request by donmccurdy**
The `@deck.gl/carto` module has gradually become more complex, with logic for CARTO's API that doesn't necessarily have parallels in other `@deck.gl/*` modules. In this PR, I'm refactoring the API interaction into a new package (`@carto/api-client`) with the intention that `@deck.gl/carto` can remain more focused on implementations of layers for deck.gl. The new dependency remains under the same open source license (MIT), and provides both ESM and CommonJS builds.

If possible, I'd like to include this change in deck.gl v9.1. It should be backward-compatible with v9.0, but it seems more appropriate to push in a minor release than a patch release.

#### Change List

- carto: Add `@carto/api-client` dependency, remove previous implementations

### CODE REVIEW

**Comment by ibgreen** [comment]
It would also be nice to start working on "graduating" the nice layers in the carto module and move them into our general layer modules.

My view is that the carto module was created partly to allow carto to iterate faster on deck.gl extensions without having to negotiate every change with maintainers. However, as useful features mature could consider moving them if they are of general interest.

If we do this would also reduce the size of the carto module helping to address the problem from another angle.

**Comment by felixpalmer** [comment]
@ibgreen agree with the long term view to move code into the `layers` module, however there is some complexity given that the layers expect a custom data format. That said, good candidates for inclusion are `PointLabelLayer` & `HeatmapTileLayer`. PRs/RFCs for these are on my list once we land 9.1

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**
